apps/server/src/trpc/list.ts

```diff
@@ -1,39 +1,88 @@
-import { flatten, validModuleRegex } from '@modtree/utils'
+import { flatten } from '@modtree/utils'
 import { z } from 'zod'
 import { createRouter } from './router'
 import { api } from '../main'
+import { base, entities } from '../schemas/entities'
+import { parseCommaSeparatedString } from '../utils/parse'
 
 export const list = createRouter()
   /** list modules */
   .query('modules', {
-    input: z.array(z.string().regex(validModuleRegex)),
+    meta: {
+      openapi: {
+        enabled: true,
+        tags: ['Module'],
+        method: 'GET',
+        path: '/modules',
+        summary: 'Get many modules',
+        description: 'Expects a comma separated string.',
+      },
+    },
+    input: z.object({
+      moduleCodes: z.string().transform(parseCommaSeparatedString),
+    }),
```

Comment:
`z.preprocess()` doesn't seem to work here
- using `z.preprocess()` generates a URL like `/modules?moduleCodes=MA2001&moduleCodes=CS1010`
- only the first variable gets captured

this method
- generates a URL like `/modules?moduleCodes=MA2001,CS1010`
- first variable gets captured, but there is only one